core/src/main/java/org/ehcache/Ehcache.java

```diff
@@ -285,9 +285,19 @@ private boolean newValueAlreadyExpired(K key, V oldValue, V newValue) {
     
     final Duration duration;
     if (oldValue == null) {
-      duration = runtimeConfiguration.getExpiry().getExpiryForCreation(key, newValue);
+      try {
+        duration = runtimeConfiguration.getExpiry().getExpiryForCreation(key, newValue);
+      } catch (RuntimeException re) {
+        logger.error("Expiry caused an exception ", re);
```

Comment:
Clarify message: _Expiry computation caused an exception - Expiry duration will be 0_